### DISCUSSION TITLE
Kabellänge DCplus & DCminus zum Wechselrichter

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#618] [Q&A]
Ich habe nicht gut aufgepasst... Ich habe gelesen,

- alle Hoymiles Eingänge sollten mit der Batterie angeschlossen sein
- alle Kabel sollen die **gleiche Länge** haben

Frage: Bedeutet dies 
- "Alle DCplus Kabel" unter sich sollen die gleiche Länge haben?
- "Alle DCminus Kabel" unter sich sollen die gleiche Länge haben?

(also, es kann sein, dass die DCplus Kabel etwas länger sind als die DCminus)

**oder**
- "Alle DCplus & DCminus Kabel" sollen die gleiche Länge haben?

Meine Logik sagt mir, so Lange die "Schleife" (plus Pol von der Batterie bis zum Wechselrichter + minus pol von der Batterie bis zum Wechselrichter) die gleiche Länge hat, die Verluste sind gleich für jeden Eingang. 
Daraufhin, wäre es kein Problem wenn die DCplus Kabellänge etwas länger ist als die DCminus.

Aber "Logik" und "Ahnungslosigkeit"  passen nicht gut zusammen, daher lieber fragen....

**Comment by SW-Niko** [comment]
Hallo @Manos1966 ,
Ich denke die leicht unterschiedlichen Verluste sind nicht das Problem. Die gemeinsame interne Masse (DCminus) der Eingänge bei einem Hoymiles schon eher. Wenn die DCminus Eingänge der Hoymiles intern verbunden sind, dann fließt der Rückstrom zur Batterie bevorzugt über die niederohmige Verbindung. Oder anders ausgedrückt in einer Rückleitung fließt mehr Strom als in der anderen Rückleitung. 
Das könnte zu einer Überlastung der internen Leiterbahnen oder der Externen Kabel führen.
Ich habe leider nur einen Hoymiles mit einem Eingang darum kann ich meine Vermutung nicht überprüfen.

**Reply by spcqike** [reply]
das sind berechtigte Bedenken. Ich selber habe noch keinen Wechselrichter geöffnet, könnte an meinem HM-600 aber bei Bedarf mal nachmessen.

demnächst bestelle ich einen HMS-2000-4T, da messe ich auch vorher mal durch.

so wie es aussieht, sind die MPPTs voneinander getrennt.
https://www.photovoltaikforum.com/thread/171257-hoymiles-hm800-leistungssteuerung-per-pwm/

Beim HM-1500 und co. sind ja immer zwei Eingänge pro MPPT, da sollten die Leitungen wohl gleich lang sein. DC+ und DC- könnte bestimmt unterschiedlich sein, wenn es denn sein muss (da man in der Regel in der +-Leitung eine Sicherung hat und auch ein BMS meistens nur in einer Leitung trennt, sind die Leitungswiderstände vom Wechselrichter bis zur eigentlichen Batterie/Zelle auf + und - so oder so unterschiedlich, oder?), aber alle DC+ Leitungen eines Eingangs (und auch die DC- Leitungen) sollten wohl die gleiche Länge / den gleichen Widerstand haben.

@Manos1966 so ganz generell, warum überhaupt unterschiedliche Leitungen? Abgesehen vom Aufbau, dass bspw. + kürzer ist weil es an eine Sicherung geht, bevor es an ne Busbar geht, und - geht eben direkt auf eine Busbar. aber es ist doch einfacher direkt 4 oder eben 8 gleich lange Kabel zu machen, oder? 😄

**Reply by Manos1966** [reply]
Danke @SW-Niko 
Mist! Google Translator hat keine Übersetzung in Deutsch geben können 🤣 

Ich glaube ich habe es verstanden 😄 

-  die DC minus&plus Eingänge der Hoymiles sind pro-MPPT intern verbunden. Zum Beispiel, ein HM-1500 hat zwei MPPTs und vier Eingaenge. Daraufhin sind Eingang 1&2 bei MPPT#1, Eingang 3&4 bei MPPT#2. Ob sie zusätzlich mit Dioden intern getrennt sind, weiss ich nicht. Da aber die meisten Solar Panele Dioden haben, sollte es nicht zusaetzlich notwendig sein gehe ich davon aus.
- So Lange die minus Kabel die gleiche Laenge haben ist alles OK (Rückstrom zur Batterie gleichmäßig über alle Verbindungen (und nicht bevorzugt über die niederohmige Verbindung))

Damit ist meine Frage beantwortet, meine minus Leitungen duerfen 40cm Lang sein, waehrend meine plus 30cm lang sind.

FRAGE: Darf ich davon ausgehen, das die gleiche Logik (Gefahr einer niederohmiger Verbindung) gilt für die Hinleitung (plus Kabel)?
Ich werde deine Erklärung in die Wiki irgendwo addieren 😉

**Reply by SW-Niko** [reply]
Hallo @spcqike ,
nach den Bilder schaut es so aus als ob die Massen der MPPT nicht verbunden sind. Aber ich wäre da vorsichtig.
Vielleicht gibt es eine innere Masselage auf der Platine oder die MPPT Massen sind über die Steuerung niederohmig verbunden.

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129541740/c40178db-4c1f-420b-9a83-7e16276ddb59)

In einem Fehlerfall (Untere Leitung hochohmig oder unterbrochen könnte der gesamte Rückstrom über eine Leitung fließen. Das wird die externe Leitung verkraften aber ob das auch die interne Verbindung schafft ist fraglich.
Falls du mal Messungen machst ... bitte informiere mich was dabei raus gekommen ist. 

@Manos1966: Vermutlich ist das bei der DC+ Leitung kein Problem weil hier die elektronische Strombegrenzung der Hoymiles greift.

**Reply by Manos1966** [reply]
> Vermutlich ist das bei der DC+ Leitung kein Problem weil hier die elektronische Strombegrenzung der Hoymiles greift.

Vielen Dank für die Erklärungen @SW-Niko !
Das die DC+ Leitung kein Problem darstellt (wahrscheinlich), sondern nur die DC-, hätte ich nicht erwartet. 
Faszinierend! Jetzt macht auch Sinn die ganze Diskussion, warum die galvanische Trennung fuer die Masse des VE.Direkt Kabels wichtig ist. [level shifter-optocoupler](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Victron-MPPT-Tipps#level-shifter---optocoupler)

"Man gibt anderen die Hand, sie nehmen den ganzen Arm" 
Ich habe noch eine brennende Frage, wenn du erlaubst:
- Ist es sinnvoll, eine FI-Sicherung vor dem Wechselrichter zu installieren?
 
Die Diskussionen, die ich bis jetzt gefunden habe, liegen 50% dafuer - 50% dagegen 🤦‍♂️ 🙄 
Vielleicht hast du eine Idee.... Falls nicht, ist es auch OK 😄

**Reply by SW-Niko** [reply]
Ein FI oder RCD in der AC Zuleitung halte ich für sehr sinnvoll wenn die Gefahr besteht das auf der Leitung zur Haussicherung jemand mit der Phase in Berührung kommt. 
Beispiel: Bei mir hängt der Inverter auf der Leitung zum Wohnzimmer (Fernseher, Soundbar, Lampe usw.). Bei einer Verbindung gegen Erde bei einem der Geräte kommt der RCD von der Hausinstallation normalerweise nach 30ms und der Inverter muss erst nach 200ms abschalten.
Diese zusätzlichen 170ms können tödlich sein.
Aktuell habe ich auch noch keinen RDC aber ich denke beim nächsten Umbau werde ich einen Einbauen.

**Reply by JaMo523** [reply]
Hast du die 30ms eventuell mit 30mA vertauscht? Im TN Netz muss die Auslösezeit für einen Fehlerstromschutzschalter unter 0,4 Sekunden liegen. Der Inverter löst also mit seinen 200 ms vermutlich schneller aus als der RCD.

**Reply by SW-Niko** [reply]
Hallo @JaMo523,
ja da habe ich etwas vertauscht. Bei mir ist ein Hager CDS 440D installiert und der hat nach Datenblatt 30mA Auslösestrom. 
Bei der Auslösezeit hatte ich 30ms im Kopf aber da lag ich falsch.
Dann ist der Inverter mit 200ms schneller .... aber nur ... wenn sie sich nicht gegenseitig beeinflussen. 
Ich vermute der HM-300 ist gegenüber den 230V galvanisch getrennt und gegen Erde so hochohmig das sowieso kein Problem für den installierten RCD besteht. Vielleicht messe ich das mal nach oder teste es mit einem RCD-Tester.
Danke für den Hinweis.

**Reply by poolcat4711** [reply]
Und bitte nicht FI hinter FI setzen. Wenn die Leitung also schon abgesichert ist, kein FI.

**Comment by Manos1966** [comment]
Hi @spcqike ich habe deine Nachricht nachdem ich beantwortet habe, gesehen.

Na ja, ich kämpfe ein bisschen mit der Planung... da ich vorsichtshalber Kühlung für Vicron und Hoymiles addieren will
[Fan-speed-controller-How-to-add-temperature-sensing-to-any-fan](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Fan-speed-controller-How-to-add-temperature-sensing-to-any-fan)
(ich sehe die Test-Installation vier Bifaziallen-Paneelen senkrecht am Zaun bring eine HM1200 auf 100% bei 3 Grad Außentemperatur! Dann könnte es im Sommer, im Keller, mit Solar-Passthrough "warm" werden...).  
Um die Luftstrome zu optimieren, will ich die Busbars an einer Seite platzieren anstatt mittig. Das gibt etwas längere DC+ Kabel als DC-

Ich bin noch am überlegen

**Reply by Manos1966** [reply]
Nachtrag (bevor jemand mich in die Luft "zerreißt" 🤣 ):
Lieber bewege ich die Busbars und dadurch mache ich die vier DC+/DC- Leitungen die zum Hoymiles fuehren, laenger, weil sie jeweils max 10,5A tragen.
Denn die anderen (Batterie -> Busbar mit 42A, MPPT-> Busbar mit 35A) deutlich hoehere Stroeme tragen.

**Reply by poolcat4711** [reply]
bei mir im keller unter voll-last keine 50°C

**Reply by Manos1966** [reply]
Ich habe meinem bei 85°C gesehen...

**Comment by spcqike** [comment]
> Die gemeinsame interne Masse (DCminus) der Eingänge bei einem Hoymiles schon eher.

hab meinen HMS-2000 nun da. Ja, die DC_minus Leitungen sind intern alle verbunden.
Bei galvanisch getrennten Stromquellen (Solarmodulen) kein Problem, wenn alle Eingänge an einer Quelle hängen (Batterie) hingegen .... Ja, da sollten die Widerstände aller DC_minus-Leitungen (vom Anschluss bis zur gemeinsamen Masse, also Busbar, Shunt, oder Batterie, je nachdem) möglichst gleich sein.

**Reply by Manos1966** [reply]
Ich habe das Thema in die Wiki hier:
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Cable-length%28s%29-between-battery-and-inverter

**Reply by Manos1966** [reply]
@spcqike wirst du ihn mit einer Pylontech verbinden, oder ueber SHUNT?

**Reply by spcqike** [reply]
vorerst, weder noch. 

erstmal je 2 Trina Vertex S+ 435W parallel pro Eingang (8 in Summe) und dann offen betreiben. Solarladeregler und Batterie kommen später.

**Reply by Manos1966** [reply]
Mist! Kannst du dich erinnern jemals gelesen zu haben, ob das SN65HVD230 CAN-Board (fuer die Pylontech) hard codiert auf SPI 2 ist? Dadurch haette es Probleme mit dem CMT2300A Board deiner HMS/HMT gehabt.... wie der Fall ist mit CMT2300A und das MCP2515 Board fuer die Huawei.
